provider/cldr/src/transform/list/mod.rs

```diff
@@ -61,14 +62,60 @@ impl DataProvider<ListFormatterPatternsV1Marker> for ListProvider {
             None => return Err(DataError::MissingResourceOptions(req.clone())),
         };
 
-        let patterns = match req.resource_path.key {
+        let mut patterns = match req.resource_path.key {
             key::LIST_FORMAT_AND_V1 => parse_and_patterns(data),
             key::LIST_FORMAT_OR_V1 => parse_or_patterns(data),
             key::LIST_FORMAT_UNIT_V1 => parse_unit_patterns(data),
-            _ => panic!("Cannot happen due to check in supports_key"),
+            _ => unreachable!(),
         }
         .map_err(DataError::new_resc_error)?;
 
+        if langid.language == langid!("es").language {
+            match req.resource_path.key {
+                // Replace " y " with " e " before /i/ sounds.
+                // https://unicode.org/reports/tr35/tr35-general.html#:~:text=important.%20For%20example%3A-,Spanish,AND,-Use%20%E2%80%98e%E2%80%99%20instead
+                key::LIST_FORMAT_AND_V1 | key::LIST_FORMAT_UNIT_V1 => patterns.replace_patterns(
+                    "{0} y {1}".parse().unwrap(),
+                    ConditionalListJoinerPattern::from_regex_and_strs(
+                        // Starts with i or (hi but not hia/hie)
+                        "i|hi([^ae]|$)",
+                        "{0} e {1}",
+                        "{0} y {1}",
+                    )
+                    .unwrap(),
+                ),
+                // Replace " o " with " u " before /o/ sound.
+                // https://unicode.org/reports/tr35/tr35-general.html#:~:text=agua%20e%20hielo-,OR,-Use%20%E2%80%98u%E2%80%99%20instead
+                key::LIST_FORMAT_OR_V1 => patterns.replace_patterns(
+                    "{0} o {1}".parse().unwrap(),
+                    ConditionalListJoinerPattern::from_regex_and_strs(
+                        // Starts with o, ho, 8 (including 80, 800, ...), or 11 either alone or followed
+                        // by thousand groups and/or decimals (excluding e.g. 110, 1100, ...)
+                        r"o|ho|8|(11(\.?\d\d\d)*(,\d*)?([^\.,\d]|$))",
+                        "{0} u {1}",
+                        "{0} o {1}",
+                    )
+                    .unwrap(),
+                ),
+                _ => unreachable!(),
+            }
+        }
+
+        if langid.language == langid!("he").language {
+            // Add dashes between ו and non-Hebrew characters
+            // https://unicode.org/reports/tr35/tr35-general.html#:~:text=is%20not%20mute.-,Hebrew,AND,-Use%20%E2%80%98%2D%D7%95%E2%80%99%20instead
+            patterns.replace_patterns(
+                "{0} \u{05D5}{1}".parse().unwrap(), // ״{0} ו {1}״
+                ConditionalListJoinerPattern::from_regex_and_strs(
+                    // Starts with a non-Hebrew letter
+                    r"[^\p{IsHebrew}]",
```

Comment:
Issue: `\p{IsHebrew}` requires Unicode property data, which we want to provide ourselves, instead of duplicating whatever the regex crate is using.

There are several ways to do this:

1. Make ListProvider depend on the raw Unicode property data providers in *provider/uprops*.
2. Make the runtime ListFormatter depend on the appropriate data key and load the data at runtime.
3. Hard-code the Hebrew regex in this function (ranges of code points), and have a unit test that ensures it remains up-to-date. The unit test could be in *experimental/list* (soon *components/list*) and check the data coming from the testdata provider.

Option 3 is probably easiest.

Reply:
Regex equality sounds brittle. I've tried to make 1. work, however I'm not sure how to obtain the correct uprops data directory.